api-reference/10 UI Components/dxDataGrid/1 Configuration/scrolling/mode.md

```diff
@@ -3,29 +3,38 @@ id: dxDataGrid.Options.scrolling.mode
+A user scrolls a single page only. Scrolling is available only if [all the page's rows](/api-reference/10%20UI%20Components/GridBase/1%20Configuration/paging/pageSize.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/paging/#pageSize') do not fit into the UI component's height. In this mode, the [pager](/api-reference/10%20UI%20Components/GridBase/1%20Configuration/pager '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/pager/') performs the main navigation and scrolling is auxiliary. If you disable the [paging](/api-reference/10%20UI%20Components/GridBase/1%20Configuration/paging '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/paging/'), the UI component loads all rows simultaneously. This action can reduce the UI component's performance. In this case, use another scrolling mode.
 
 - **Virtual**       
-Rows are loaded when they get into the viewport and removed once they leave it. In this mode, users can move to any page instantly.
-
-    If the rows take time to be loaded and rendered, they display gray boxes. [Rendering optimization](/concepts/05%20UI%20Components/DataGrid/10%20Enhance%20Performance%20on%20Large%20Datasets/030%20Rendering%20Optimization.md '/Documentation/Guide/UI_Components/DataGrid/Enhance_Performance_on_Large_Datasets/#Rendering_Optimization') can reduce rendering time and remove the gray boxes. Alternatively, you can disable the **scrolling**.[renderAsync](/api-reference/10%20UI%20Components/GridBase/1%20Configuration/scrolling/renderAsync.md '{basewidgetpath}/Configuration/scrolling/#renderAsync') property, but this can impact performance.
+The component loads pages when they get into the viewport and removes them once they leave it. In this mode, users can move to any page instantly. If the component takes time to load and render rows, they display gray boxes. [Rendering optimization](/concepts/05%20UI%20Components/DataGrid/10%20Enhance%20Performance%20on%20Large%20Datasets/030%20Rendering%20Optimization.md '/Documentation/Guide/UI_Components/DataGrid/Enhance_Performance_on_Large_Datasets/#Rendering_Optimization') can reduce rendering time and remove the gray boxes. Alternatively, you can disable the **scrolling**.[renderAsync](/api-reference/10%20UI%20Components/GridBase/1%20Configuration/scrolling/renderAsync.md '{basewidgetpath}/Configuration/scrolling/#renderAsync') property, but this can impact performance.
```

Comment:
1. `The component loads pages when they get into the viewport and removes them once they leave it.`

a) Are you sure "pages" is correct? Or do you mean "rows"?

b) Can we change "when they get into the viewport" to "while they are visible within the viewport"?

2. `[Rendering optimization](/concepts/05%20UI%20Components/DataGrid/10%20Enhance%20Performance%20on%20Large%20Datasets/030%20Rendering%20Optimization.md '/Documentation/Guide/UI_Components/DataGrid/Enhance_Performance_on_Large_Datasets/#Rendering_Optimization') can reduce rendering time and remove the gray boxes.`

Can we change "rendering time" to "render time"?

Reply:
1. a) Customers often think that virtual scrolling only affects how rows are rendered. However, it is very important that virtual/infinite scrolling always loads data by pages. Even if paging.enabled is set to false, this setting will be ignored.

Reply:
I think the way you wrote it ("The component loads pages...") is sufficient to emphasize that virtual scrolling loads data by page. Thanks for the clarification.